patches/truffle+5.1.26.patch

```diff
@@ -10,3 +14,17 @@ index 5991472..eab1676 100755
    const outputString = solc.compile(inputString);
    const compilerOutput = JSON.parse(outputString);
  
+@@ -203423,6 +203424,13 @@ const reason = {
+       res && typeof res === "object" && res.error && res.error.data;
+     const isString =
+       res && typeof res === "object" && typeof res.result === "string";
++    const isHardHat =
++      res && typeof res === "object" && res.error && res.error.message && !res.error.data;
++
++    if (isHardHat) {
++      if (res.error.message.includes('Transaction reverted without a reason string')) return 'revert'
++      return res.error.message.match('VM Exception while processing transaction: reverted with reason string \'(.*)\'')[1]
++    }
```

Comment:
can you elaborate and write a very descriptive comment in this patchfile? 
I was hoping we can get rid of patchfiles one day...

Reply:
On eth_call, ganache returns error object that has message, code, data fields. Truffle reads the revert reason from res.error.data in this case. But hardhat returns error object only with message and code, so to return the revert reason, I needed to parse res.error.message.

In general, when eth_call returns an error, the spec doesn't specify a data field containing the revert reason, so truffle is tailored specifically for ganache here

Reply:
can you please write this comment in the patch itself, maybe? it will be impossible to remember few months later